closecloser/ignored.go

```diff
@@ -78,3 +79,25 @@ type fun struct {
 	pkg  string
 	name string
 }
+
+func isAllowedType(ty types.Type) bool {
+	if ty == nil {
+		return false
+	}
+	if p, ok := ty.(*types.Pointer); ok {
+		return isAllowedType(p.Elem())
+	}
+	if n, ok := ty.(*types.Named); ok {
+		for _, m := range allowedTypes {
+			if m.Matches(n) {
+				return true
+			}
+		}
+		return false
+	}
+	return false
+}
```

Comment:
You could consider a type check switch here


```suggestion
func isAllowedType(ty types.Type) bool {
	switch typ := tyt.(type){
		case *types.Pointer:
			return isAllowedType(typ.Elem())
	
		case *types.Named:
			for _, m := range allowedTypes {
				if m.Matches(typ) {
					return true
				}
			}
	}
	return false
}
```

Not only it's more readable, but the performance are better because the type assertion is only done once, and not twice

Reply:
Thanks. I considered `switch` and wondered which is better when I wrote it. I've chosen `if` intentionally, but don't have clear justification. Anyway let me go with `if`, that I prefer.